cmse/cmse.md

```diff
@@ -1842,29 +1846,20 @@ __acle_se_foo:
     bl      bar
     @11: restore used callee-saved registers
     pop     {r4-r6, lr}
-    @12: if called from secure, we are done
+    @12: restore FPCXTNS if called from non-secure
     tst     lr, #1
-    it      ne
+    itte    eq
+    popeq   {r12, lr}
+    vmsreq  fpcxtns, r12
+    @12: if called from secure, we are done
     bxne    lr
     @13: pop secure stack space
     add     sp, sp, #8
-    @14: check SFPA bit to see if FP is used
-    mrs     r1, control
-    tst     r1, #8
-    bne     .LdoneFP
-    @15: clear floating point caller-saved registers
-    mov     r1, #0
-    vmov    s0, s1, r1, r1
-    vmov    s2, s3, r1, r1
-    ...
-    vmov    s30, s31, r1, r1
-    @16: clear floating point flags
-    vmsr    fpscr, r1
+    @14: clear floating point caller-saved registers if a FP context is active
+    vscclrm {s0-s31}
 .LdoneFP:
-    @17: clear integer caller-saved registers except for return value
-    mov     r1, #0
-    mov     r2, #0
-    mov     r3, #0
+    @15: clear integer caller-saved registers except for return value
+    clrm    {r1-r3}
     @18: clear other registers and the integer status flags
     mov     r12, #0
```

Comment:
both of these can also be done by the clrm, just add APSR and R12 to the register list